### PR TITLE
#709,#861 - Make the REF_DIR configurable in plugin installation scripts

### LINKED ISSUE #861: copy_references: Add support of custom /usr/share/jenkins/ref/ locations
Currently `copy_references` method in jenkins-support hardcodes `/usr/share/jenkins/ref/` by making substraction at 23 symbol in the codebase. It would be great to make /usr/share/jenkins/ref/  configurable so that custom images can specify other reference paths

### PR DESCRIPTION
**Pull request by slide**
This allows overriding the default REF dir

### CODE REVIEW

File: jenkins-support [hunk 1]
```diff
@@ -40,7 +42,7 @@ get_plugin_version() {
 copy_reference_file() {
     f="${1%/}"
     b="${f%.override}"
-    rel="${b:23}"
+    rel="${b#"$REF/"}"
```
Comment (oleg-nenashev) [inline jenkins-support]: There are also references below

Comment (slide) [reply jenkins-support]: Fixed by making `dir` use `rel` instead of `b` and removing the offset.

**Comment by oleg-nenashev** [comment]
In my local patch I used a new `reldir` variable to just replace the substitutions. Should have submitted the patch, but it was rather a hack&slash

**Comment by slide** [comment]
The `dir` variable is only used in context of a relative path to `JENKINS_HOME` so I don't think a new variable is needed.

**Comment by oleg-nenashev** [comment]
My fix is in https://github.com/cloudbees-oss/cjd-jcasc-demo/blob/master/jenkins-support . Sorry for not having the diff for the change, I was planning to fix it immediately after reporting :(

**Comment by oleg-nenashev** [comment]
Would be great to have some docs also

**Comment by slide** [comment]
Good idea

**Comment by oleg-nenashev** [comment]
Retriggering CI

**Comment by slide** [comment]
I'm back from travel today, so I should be able to look at the requested changes.

**Comment by MRamonLeon** [comment]
Seems like there is no space to run docker:

> docker: Error response from daemon: mkdir /var/lib/docker/overlay2/85a56d000bfdfe21cbfef7476d59ac983f1db2c6fcebe5a41c241b4e08967d05-init: no space left on device.

File: tests/install-plugins/ref/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,3 @@
+FROM bats-jenkins-install-plugins-ref
```
Comment (MRamonLeon) [inline tests/install-plugins/ref/Dockerfile]: I needed this change to be able to run the test successfully
```suggestion
FROM bats-jenkins-install-plugins
```

**Comment by MRamonLeon** [comment]
If the meaning of the REF param changes, the documentation from my PR needs to be updated: 
* https://github.com/jenkinsci/docker/pull/873/files#diff-04c6e90faac2675aa89e2176d2eec7d8R223
* https://github.com/jenkinsci/docker/pull/873/files#diff-b1bbc6bb35d75111f8f537a2b5ecc63bR9

**Comment by slide** [comment]
The meaning of `REF` is not changing, but your description is not complete. It is not just for plugins, it can contain anything you want to pre-seed to get copied into `JENKINS_HOME`. You can put init.groovy.d scripts in there, etc. 

I will fix the test, not sure why I had the wrong image in there.

**Comment by MRamonLeon** [comment]
> The meaning of `REF` is not changing, but your description is not complete. It is not just for plugins, it can contain anything you want to pre-seed to get copied into `JENKINS_HOME`. You can put init.groovy.d scripts in there, etc.

Maybe I was a bit confused with https://github.com/jenkinsci/docker/pull/866/files#diff-b1bbc6bb35d75111f8f537a2b5ecc63bL10

I still see something, now if **REF** is not set, **REF_DIR** is /plugins, instead of /usr/share/jenkins/ref/plugins. Not sure if it's possible.

**Comment by slide** [comment]
Yes, I need to add REF to the Dockerfile with a default value so it is always set to something. Good catch

**Comment by slide** [comment]
@oleg-nenashev It looks like @MRamonLeon has some of the documentation stuff in his PR, would that be ok instead of adding it in this PR?

**Comment by oleg-nenashev** [comment]
Fine with me

**Comment by oleg-nenashev** [comment]
I am testing it locally to include it into the next weekly.
CI for docker is still dead, see https://groups.google.com/forum/#!topic/jenkinsci-dev/v9d-XosOp2s

**Pull request merged**